qa/Rakefile

```diff
@@ -27,21 +27,21 @@ namespace :test do
   namespace :acceptance do
     desc "Run all acceptance"
     task :all do
-      exit(RSpec::Core::Runner.run([Rake::FileList["acceptance/spec/**/*_spec.rb"]]))
+      exit(RSpec::Core::Runner.run([Rake::FileList["acceptance/spec/lib/*_spec.rb"]]))
     end
 
     platforms.types.each do |type|
       desc "Run acceptance test in #{type} machines"
       task type do
-        exit(RSpec::Core::Runner.run([Rake::FileList["acceptance/spec/#{type}/**/*_spec.rb"]]))
+        ENV['LS_TEST_PLATFORM']=type
+        exit(RSpec::Core::Runner.run([Rake::FileList["acceptance/spec/lib/*_spec.rb"]]))
       end
     end
 
     desc "Run one single machine acceptance test"
     task :single, :machine do  |t, args|
-      ENV['LS_VAGRANT_HOST'] = args[:machine]
-      platform = LogStash::VagrantHelpers.translate(args[:machine])
-      exit(RSpec::Core::Runner.run([Rake::FileList["acceptance/spec/**/*_spec.rb"]]))
+      ENV['LS_VAGRANT_HOST']  = args[:machine]
+      exit(RSpec::Core::Runner.run([Rake::FileList["acceptance/spec/lib/*_spec.rb"]]))
     end
   end
 end
```

Comment:
Not implemented here, but i certainly can add it.
   /purbon

On Fri, 29 Apr 2016 21:07 Pier-Hugues Pellerin, <notifications@github.com>
wrote:

> In qa/Rakefile
> <https://github.com/elastic/logstash/pull/5226#discussion_r61630135>:
>
> >        end
> >      end
> >
> >      desc "Run one single machine acceptance test"
> >      task :single, :machine do  |t, args|
> > -      ENV['LS_VAGRANT_HOST'] = args[:machine]
> > -      platform = LogStash::VagrantHelpers.translate(args[:machine])
> > -      exit(RSpec::Core::Runner.run([Rake::FileList["acceptance/spec/**/*_spec.rb"]]))
> > +      ENV['LS_VAGRANT_HOST']  = args[:machine]
> > +      exit(RSpec::Core::Runner.run([Rake::FileList["acceptance/spec/lib/*_spec.rb"]]))
> >      end
> >    end
> >  end
>
> @purbon <https://github.com/purbon> How can I run one specific example
> from a specific file, I believe this is a really important use case since
> running the test takes a lot of time.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/pull/5226/files/8eda756fa28d4a18a18e8dcf4e0df3b57b64bf36#r61630135>
>